### ISSUE TITLE
Translate Sponge's strings

### ISSUE THREAD

**Issue by zml2008** [SpongePowered/SpongeAPI#564]
The end goal of this is to make Sponge fully internationalized, meaning that server owners get messages all in their own language!

- [ ] Make text API support serverside translations -- PR #563 
- [ ] Add GettextTranslation -- Gettext has better tooling
- [ ] Write Gradle plugin to use gettext to extract strings from translation methods
- [ ] Make every string in Sponge (API + mod) translatable
- [ ] Setup CrowdIn for people to translate everything

**Comment by gabizou** [comment]
> Make every string in Sponge (API + mod) translatable

Some things (like DataQuery keys) obviously should not be translatable. I hope you weren't considering that as a possibility.

**Comment by zml2008** [comment]
yeah, not those. more user-facing messages was what I'm talking about. updated the issue to be more clear about that.

**Comment by Zidane** [comment]
@zml2008 
Status update on this.

**Comment by gabizou** [comment]
This needs further thought.

**Comment by ST-DDT** [comment]
This could be implemented using the features added in #808 and its prior PR, but that would require changes to the Translation class.
Namely changing `get(Locale locale, Object... args) : Text` (instead of `String`) and adding a `resolveLocally()` method.

And on the implementation side.

`CommandSource`
````java
public void sendMessage(Text text) {
    if (text instanceOf Text.Translatable) {
        if (text.getTranslation().resolveLocally()) {
            text = text.getTranslation().get(this.getLocale(), text.getArguments()); 
            // Internally calling i.e. Text.format(dictionary.getTranslation(getId(), locale), args);
        }
    }
    [...sendToClient...]
}
````

I already thought about adding that feature to my PR #808, but this would be outside of its scope.

**Comment by zml2008** [comment]
@ST-DDT none of that is needed -- this works as is today, just needs the proper tooling (a gradle plugin) for gettext to generate pot files and the appropriate resource bundles.

**Comment by gravityfox** [comment]
do you think this could be expanded to an api that allows plugins to easily support translations as well?

**Comment by ryantheleach** [comment]
We are hoping to finally have the Text Implementation internal to Sponge come API 8, which will pave the way to this and other translations working nicely finally.
